src/Illuminate/Auth/Guard.php

```diff
@@ -9,755 +11,780 @@
+     */
+    protected function getRecaller()
+    {
+        return $this->request->cookies->get($this->getRecallerName());
+    }
+
+    /**
+     * Get the user ID from the recaller cookie.
+     *
+     * @return string
+     */
+    protected function getRecallerId()
+    {
+        if ($this->validRecaller($recaller = $this->getRecaller())) {
+            return head(explode('|', $recaller));
+        }
+    }
+
+    /**
+     * Determine if the recaller cookie is in a valid format.
+     *
+     * @param string $recaller
+     *
+     * @return bool
+     */
+    protected function validRecaller($recaller)
+    {
+        if (! is_string($recaller) || ! str_contains($recaller, '|')) {
+            return false;
+        }
+
+        $segments = explode('|', $recaller);
+
+        return count($segments) == 2 && trim($segments[0]) !== '' && trim($segments[1]) !== '';
+    }
+
+    /**
+     * Log a user into the application without sessions or cookies.
+     *
+     * @param array $credentials
+     *
+     * @return bool
+     */
+    public function once(array $credentials = [])
+    {
+        if ($this->validate($credentials)) {
+            $this->setUser($this->lastAttempted);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    /**
+     * Validate a user's credentials.
+     *
+     * @param array $credentials
+     *
+     * @return bool
+     */
+    public function validate(array $credentials = [])
+    {
+        return $this->attempt($credentials, false, false);
+    }
+
+    /**
+     * Attempt to authenticate using HTTP Basic Auth.
+     *
+     * @param string $field
+     *
+     * @return \Symfony\Component\HttpFoundation\Response|null
+     */
+    public function basic($field = 'email')
+    {
+        if ($this->check()) {
+            return;
+        }
+
+        // If a username is set on the HTTP basic request, we will return out without
+        // interrupting the request lifecycle. Otherwise, we'll need to generate a
+        // request indicating that the given credentials were invalid for login.
+        if ($this->attemptBasic($this->getRequest(), $field)) {
+            return;
+        }
+
+        return $this->getBasicResponse();
+    }
+
+    /**
+     * Perform a stateless HTTP Basic login attempt.
+     *
+     * @param string $field
+     *
+     * @return \Symfony\Component\HttpFoundation\Response|null
+     */
+    public function onceBasic($field = 'email')
+    {
+        if (! $this->once($this->getBasicCredentials($this->getRequest(), $field))) {
+            return $this->getBasicResponse();
+        }
+    }
+
+    /**
+     * Attempt to authenticate using basic authentication.
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param string                                    $field
+     *
+     * @return bool
+     */
+    protected function attemptBasic(Request $request, $field)
+    {
+        if (! $request->getUser()) {
```

Comment:
The exclamation point is not reglemented on PSR-2, I think it's more readable to keep a space before and after it too.

What do you think @GrahamCampbell